### DISCUSSION TITLE
Index From index

### DISCUSSION THREAD

**Discussion by MohMehrnia** [ravendb/ravendb#19340] [General]
I have a bellow map/reduce  index

```
//map
from v in docs.VideoPlayUserInfos
select new {
    v.CourseID, 
    v.FileID, 
    v.UserID, 
    v.FullName, 
    v.CurrentTime, 
    ContentDuration = v.ContentDuration.TotalSeconds
} 
```

//reduce 
```
from r in results
group r by new { r.UserID, r.FullName,r.CourseID, r.ContentDuration, r.FileID } into g
select new {
   g.Key.UserID,
   g.Key.FullName,
   g.Key.CourseID,
   g.Key.ContentDuration,
   g.Key.FileID,
   CurrentTime = g.Max(x=>x.CurrentTime)
}
```

and I need sum currenttime field from result with group by userid

how can create index from another index

**Comment by ppekrol** [comment]
HI, there is no ability to index directly the data from the index, but you can save those results as artificial documents and then create a secondary index based on them. More info here: https://ravendb.net/docs/article-page/6.2/Csharp/indexes/map-reduce-indexes#reduce-results-as-artificial-documents

Or maybe in your case another map-reduce index would be enough? Sth like:

```
//map
from v in docs.VideoPlayUserInfos
select new {
    v.UserID, 
    v.CurrentTime,
} 

//reduce

from r in results
group r by new { r.UserID } into g
select new {
   g.Key.UserID,
   CurrentTime = g.Sum(x=>x.CurrentTime)
}
```

**Reply by MohMehrnia** [reply]
I must get max first and then sum of max value
